### PR TITLE
DCMAW-7696: Add android network module to common repository

### PR DESCRIPTION
**Pull request by chrisdondig**
- Generic http client
- Ktor implementation of generic http client
- Online checker
- Request and Response classes

### CODE REVIEW

File: git-conventional-commits.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  commitIgnoreRegexPattern: "^WIP "
+  headlines:
+    feat: Features
+    fix: Bug Fixes
+    perf: Performance Improvements
+    merge: Merges
+    breakingChange: BREAKING CHANGES
+
+  ## GitHub
+  commitUrl: https://github.com/govuk-one-login/mobile-android-ui/commit/%commit%
```
[the remark this answers is not in the document]
Comment (chrisdondig) [reply git-conventional-commits.yaml:36]: Done

File: network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import uk.gov.android.network.api.ApiResponse
+import uk.gov.android.network.client.HttpStatusCodeExtensions.TransportError
+import uk.gov.android.network.useragent.UserAgentGenerator
+
+@Suppress("TooGenericExceptionCaught", "OptionalWhenBraces")
+class KtorHttpClient(
+    userAgentGenerator: UserAgentGenerator
+) : GenericHttpClient {
+
+    private var httpClient: HttpClient = makeHttpClient(userAgentGenerator)
```
[the remark this answers is not in the document]
Comment (chrisdondig) [reply network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt:35]: That's not possible to do without adding makeHttpClient to the GenericHttpClient interface which couples it to Ktor.

File: network/src/test/java/uk/gov/android/network/ExampleUnitTest.kt
[the remark this answers is not in the document]
Comment (chrisdondig) [reply network/src/test/java/uk/gov/android/network/ExampleUnitTest.kt:1]: Agree, removed

File: network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                        ?: return@handleResponseExceptionWithRequest
+                    val exceptionResponse = responseException.response
+
+                    throw ResponseException(exceptionResponse, exceptionResponse.bodyAsText())
+                }
+            }
+        }
+    }
+
+    override suspend fun makeRequest(apiRequest: ApiRequest): ApiResponse {
```
[the remark this answers is not in the document]
Comment (chrisdondig) [reply network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt:80]: I think a Flow is more appropriate for streams of data. The calling class could emit the ApiResponse as a flow if desired.

File: network/src/main/java/uk/gov/android/network/useragent/UserAgentGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+     * @param appName Name of the app as displayed in the play store
+     * @param versionName Version of the app
+     * @param manufacturer Manufacturer of the device running the app
+     * @param model Model number of the device running the app
+     * @param sdkVersion The current version of the Android OS
+     * @param clientName The name of the third party HTTP client
+     * @param clientVersion Version number of the HTTP client
+     */
+    fun setUserAgent(
+        appName: String,
```
[the remark this answers is not in the document]
Comment (chrisdondig) [reply network/src/main/java/uk/gov/android/network/useragent/UserAgentGenerator.kt]: Definitely, done

File: network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt:80]: Ok, so it seems then that maybe the `ApiResponse.Loading` is a little unnecessary?

[the remark this answers is not in the document]
Comment (alex-bradbury) [reply network/src/main/java/uk/gov/android/network/client/KtorHttpClient.kt:80]: ok

**Review by alex-bradbury — APPROVED** [review]

**Pull request merged**